src/mining_minigame.c

```diff
@@ -0,0 +1,3395 @@
+#include "m4a.h"
+#include "menu_helpers.h"
+#include "menu.h"
+#include "malloc.h"
+#include "scanline_effect.h"
+#include "sprite.h"
+#include "constants/rgb.h"
+#include "decompress.h"
+#include "constants/songs.h"
+#include "sound.h"
+#include "sprite.h"
+#include "string_util.h"
+#include "pokemon_icon.h"
+#include "graphics.h"
+#include "data.h"
+#include "pokedex.h"
+#include "gpu_regs.h"
+#include "random.h"
+#include "field_message_box.h"
+#include "constants/items.h"
+#include "item.h"
+#include "data/mining_minigame.h"
+
+/* >> Specials << */
+void StartMining(void);
+
+/* >> Callbacks << */
+static void Mining_Init(MainCallback callback);
+static void Mining_SetupCB(void);
+static bool8 Mining_InitBgs(void);
+static void Mining_MainCB(void);
+static void Mining_VBlankCB(void);
+
+/* >> Tasks << */
+static void Task_Mining_WaitFadeAndBail(u8 taskId);
+static void Task_MiningWaitFadeIn(u8 taskId);
+static void Task_WaitButtonPressOpening(u8 taskId);
+static void Task_MiningMainInput(u8 taskId);
+static void Task_MiningFadeAndExitMenu(u8 taskId);
+static void Task_MiningPrintResult(u8 taskId);
+
+/* >> Others << */
+static void Mining_FadeAndBail(void);
+static bool8 Mining_LoadBgGraphics(void);
+static void Mining_LoadSpriteGraphics(void);
+static void Mining_FreeResources(void);
+static void Mining_UpdateStressLevel(void);
+static void Mining_UpdateTerrain(void);
+static void Mining_DrawRandomTerrain(void);
+static void DoDrawRandomItem(u8 itemStateId, u8 itemId);
+static void DoDrawRandomStone(u8 itemId);
+static bool32 DoesStoneFitInItemMap(u8 itemId);
+static bool32 CanStoneBePlacedAtXY(u32 x, u32 y, u32 itemId);
+static void Mining_CheckItemFound(void);
+static void PrintMessage(const u8 *string);
+static void InitMiningWindows(void);
+static bool32 IsStressLevelMax(void);
+static void EndMining(u8 taskId);
+static u32 ConvertLoadGameStateToItemIndex(void);
+static void GetItemOrPrintError(u8 taskId, u32 itemIndex, u32 itemId);
+static void CheckItemAndPrint(u8 taskId, u32 itemIndex, u32 itemId);
+static void MakeCursorInvisible(void);
+static void HandleGameFinish(u8 taskId);
+static void PrintItemSuccess(u32 buriedItemsIndex);
+static u32 GetTotalNumberOfBuriedItems(void);
+static void InitBuriedItems(void);
+static bool32 AreAllItemsFound(void);
+static void SetBuriedItemsId(u32 index, u32 itemId);
+static void SetBuriedItemStatus(u32 index, bool32 status);
+static u32 GetBuriedBagItemId(u32 index);
+static u32 GetBuriedMiningItemId(u32 index);
+static u32 GetNumberOfFoundItems(void);
+static bool32 GetBuriedItemStatus(u32 index);
+static void ExitMiningUI(u8 taskId);
+static void WallCollapseAnimation();
+
+/* >> Debug << */
+static u32 Debug_SetNumberOfBuriedItems(u32 rnd);
+static u32 Debug_DetermineStoneSize(u32 stone, u32 stoneIndex);
+static void Debug_DetermineLocation(u32* x, u32* y, u32 item);
+static void Debug_RaiseSpritePriority(u32 spriteId);
+#if DEBUG_ENABLE_ITEM_GENERATION_OPTIONS == TRUE
```

Comment:
```suggestion
#if DEBUG_ENABLE_ITEM_GENERATION_OPTIONS
```
The `== TRUE` isn't necessary